pwa-devdocs/src/tutorials/extension-development/taglist/index.md

```diff
@@ -0,0 +1,474 @@
+touch src/Taglist/tag.css && \
+touch src/Taglist/tagList.js && \
+touch src/Taglist/tagList.css
+```
+
+### `index.js`
+
+Exporting modules from your component's `index.js` file is a common standard in a React project.
+It lets another component import the tag list component using the `TagList` directory name.
+
+```js
+/* src/TagList/index.js */
+
+export {default} from './tagList'
+```
+
+### `tag.js` and `tag.css`
+
+The `tag.js` and `tag.css` files define a Tag component that renders a single tag in the tag list.
+
+```jsx
+/* src/TagList/tag.js */
+
+import React from "react";
+import Button from '@magento/venia-ui/lib/components/Button'
+import { Link } from '@magento/venia-ui/lib/drivers';
+
+import classes from './tag.css'
+
+const categoryUrlSuffix = '.html'
+
+// This is a component responsible for rendering a single tag
+const Tag = (props) => {
+  // Destructures the props object into variables
+  const { value } = props;
+
+  const { name, url_path } = value;
+  const url = `/${url_path}${categoryUrlSuffix}`;
+
+  const buttonClasses = {
+        root_lowPriority: classes.root,
+        content: classes.content
+    }
+
+  return (
+    <Link className={classes.link} to={url}>
+      <Button classes={buttonClasses} priority="low" type="button">
+        {name}
+      </Button>
+    </Link>
+  );
+};
+
+// Make this function the default exported of this module
+export default Tag;
+```
+
+```css
+/* src/TagList.tag.css */
+
+.root {
+    border: solid 1px #2680eb;
+    padding: 3px 9px;
+    margin: 5px;
+    border-radius: 6px;
+}
+
+.content {
+    color: #2680eb;
+    font-size: 0.875rem;
+}
+```
+
+### `tagList.js` and `tagList.css`
+
+The `tagList.js` and `tagList.css` files define the main TagList component this package provides.
+It accepts a `categoriesListData` object as a prop and renders the data as a tag list.
+
+```jsx
+/* src/TagList/tagList.js */
+
+import React from "react";
+import Tag from "./tag";
+
+import classes from './tagList.css';
+
+const TagList = (props) => {
+  // Destructures the props object into variables
+  const { categoriesListData } = props;
+
+  const { categories } = categoriesListData;
+
+  // Returns nothing if there are no categories
+  if(!categories){
+      return null;
+  }
+
+  // Converts the array of tag strings into an array of Tag components
+  const tagList = categories.map((category) => {
+    return <Tag key={category.name} value={category} />;
+  });
+
+  // Returns the list of Tag components inside a div container
+  return <div className={classes.root}>{tagList}</div>;
+};
+
+export default TagList;
+```
+
+```css
+/* src/TagList/tagList.css */
+
+.root {
+    display: flex;
+    flex-wrap: wrap;
+}
+```
+
+### Add TagList component dependencies
+
+The TagList component requires third party libraries, such as React, to render the correct HTML.
+Since this package is an extension, you should list these as peer dependencies in your `package.json` file.
```

Comment:
```suggestion
The TagList component requires third party libraries, such as React, to render the correct HTML.
Since your package is an extension, you should list these as peer dependencies in your `package.json` file.
```

Clears up a little ambiguity 👍